### PR TITLE
fix: update cursorButton once freedraw is released

### PR DESCRIPTION
**Pull request by valgussev**
We were not correctly updating cursorButton value to "up" once freedraw is released. 

At `onPointerUpFromPointerDownHandler`:
1. We [update](https://github.com/excalidraw/excalidraw/blob/9ba7ca38456094e699c885a37e800cbb537ee870/src/components/App.tsx#L4242) cursorButton to "up"
2. Below we [call](https://github.com/excalidraw/excalidraw/blob/9ba7ca38456094e699c885a37e800cbb537ee870/src/components/App.tsx#L4331) actionFinalize which using an old state value overrides cursorButton back to "down".

This PR ensures actionFinalize is called only after state is updated.

### CODE REVIEW

**Comment by valgussev** [comment]
Force pushing the change to get lint and test checks reported

**Comment by dwelle** [comment]
Thanks. I've taken a look and decided to do fix this differently by resetting the state in the finalize action — this will fix even more cases. Also doing the same on `restore()`.

> Force pushing the change to get lint and test checks reported, is there a known issue?

It just requires a maintainer approval to run GH actions when a first-time collaborator creates a PR.

**Review by dwelle — APPROVED** [review]

**Comment by valgussev** [comment]
> fix this differently by resetting the state in the finalize action — this will fix even more cases

thanks, that is even better!

**Pull request merged**
